src/abstractdataframe/subset.jl

```diff
@@ -180,7 +196,11 @@ Each argument passed in `args` can be either a single column selector or a
 described for [`select`](@ref).
 
 Note that as opposed to [`filter!`](@ref) the `subset!` function works on whole
-columns (or all rows in groups for `GroupedDataFrame`).
+columns (or all rows in groups for `GroupedDataFrame`) and must return a vector.
+If you want to filter rows of an `AbstractDataFrame` or groups of
+`GroupedDataFrame` by a function that returns `Bool` value use [`filter!`](@ref)
+or [`filter`](@ref) (note that [`filter!`](@ref) is not supported for
+`GroupedDataFrame`).
```

Comment:
Sync this with `subset` docstring.

Reply:
I will sync once we finalize `subset` docstring.

Reply:
done